Infrastructure/Services/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Application.Interfaces.IRepositories;
+using Domain.Models;
+using Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Services.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : class, IEntity
+{
+    protected DataContext _context;
+    private DbSet<T?> dbSet;
```

Comment:
Just `DbSet<T>`. And it's a naming convention - start all local fields with underscore - `_dbSet`.